x-pack/plugins/apm/public/context/url_params_context/helpers.ts

```diff
@@ -26,13 +27,27 @@ export function getDateRange({
   rangeFrom?: string;
   rangeTo?: string;
 }) {
+  // If the previous state had the same range, just return that instead of calculating a new range.
   if (state.rangeFrom === rangeFrom && state.rangeTo === rangeTo) {
     return { start: state.start, end: state.end };
   }
 
+  const start = getParsedDate(rangeFrom);
+  const end = getParsedDate(rangeTo, { roundUp: true });
+
+  // `getParsedDate` will return undefined for invalid or empty dates. We return
+  // the previous state if either date is undefined.
+  if (!start || !end) {
+    return { start: state.start, end: state.end };
+  }
+
+  // Calculate ticks for the time ranges to produce nicely rounded values
+  const ticks = scaleTime().domain([start, end]).ticks();
```

Comment:
Looks like this doesn't include ticks from outside of the domain. From the tests you can see that the min/max from the ticks result in a smaller range, which IMHO should not happen. Maybe `nice()`ing the domain here will help: https://observablehq.com/@d3/scale-ticks.